cdap-api/src/main/java/co/cask/cdap/internal/schedule/constraint/ConcurrencyConstraint.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.schedule.constraint;
+
+/**
+ * A constraint which dictates an upper bound on the number of concurrent schedule runs.
+ */
+public class ConcurrencyConstraint extends Constraint {
```

Comment:
So you have two different set of Constraint classes? One set in API, the other in app-fabric? Maybe better to call the one in api differently? E.g. `ScheduleConstraint` and `ScheduleTrigger`?